ledger/txtail.go

```diff
@@ -196,6 +199,9 @@ func (t *txTail) newBlock(blk bookkeeping.Block, delta ledgercore.StateDelta) {
 		return
 	}
 
+	t.tailMu.Lock()
```

Comment:
nit: is it to protect `t.putLV` below? then move it closer to the loop.

Reply:
This should move up, since it needs to also protect `t.recent[rnd]`.